ledger/ledger-api-tests/infrastructure/src/main/scala/com/daml/ledger/api/testtool/infrastructure/participant/SingleParticipantTestContext.scala

```diff
@@ -0,0 +1,846 @@
+  PartyAllocationConfiguration,
+}
+import com.daml.ledger.api.v1.active_contracts_service.{
+  GetActiveContractsRequest,
+  GetActiveContractsResponse,
+}
+import com.daml.ledger.api.v1.admin.config_management_service.{
+  GetTimeModelRequest,
+  GetTimeModelResponse,
+  SetTimeModelRequest,
+  SetTimeModelResponse,
+  TimeModel,
+}
+import com.daml.ledger.api.v1.admin.package_management_service.{
+  ListKnownPackagesRequest,
+  PackageDetails,
+  UploadDarFileRequest,
+}
+import com.daml.ledger.api.v1.admin.participant_pruning_service.{PruneRequest, PruneResponse}
+import com.daml.ledger.api.v1.admin.party_management_service.{
+  AllocatePartyRequest,
+  GetParticipantIdRequest,
+  GetPartiesRequest,
+  ListKnownPartiesRequest,
+  PartyDetails,
+}
+import com.daml.ledger.api.v1.command_completion_service.{
+  Checkpoint,
+  CompletionEndRequest,
+  CompletionEndResponse,
+  CompletionStreamRequest,
+  CompletionStreamResponse,
+}
+import com.daml.ledger.api.v1.command_service.{
+  SubmitAndWaitForTransactionIdResponse,
+  SubmitAndWaitForTransactionResponse,
+  SubmitAndWaitForTransactionTreeResponse,
+  SubmitAndWaitRequest,
+}
+import com.daml.ledger.api.v1.command_submission_service.SubmitRequest
+import com.daml.ledger.api.v1.commands.{Command, Commands, ExerciseByKeyCommand}
+import com.daml.ledger.api.v1.completion.Completion
+import com.daml.ledger.api.v1.event.Event.Event.Created
+import com.daml.ledger.api.v1.event.{CreatedEvent, Event}
+import com.daml.ledger.api.v1.ledger_configuration_service.{
+  GetLedgerConfigurationRequest,
+  GetLedgerConfigurationResponse,
+  LedgerConfiguration,
+}
+import com.daml.ledger.api.v1.ledger_offset.LedgerOffset
+import com.daml.ledger.api.v1.package_service._
+import com.daml.ledger.api.v1.testing.time_service.{GetTimeRequest, GetTimeResponse, SetTimeRequest}
+import com.daml.ledger.api.v1.transaction.{Transaction, TransactionTree}
+import com.daml.ledger.api.v1.transaction_filter.{Filters, InclusiveFilters, TransactionFilter}
+import com.daml.ledger.api.v1.transaction_service.{
+  GetLedgerEndRequest,
+  GetTransactionByEventIdRequest,
+  GetTransactionByIdRequest,
+  GetTransactionsRequest,
+  GetTransactionsResponse,
+}
+import com.daml.ledger.api.v1.value.{Identifier, Value}
+import com.daml.ledger.client.binding.Primitive.Party
+import com.daml.ledger.client.binding.{Primitive, Template}
+import com.daml.lf.data.Ref
+import com.daml.logging.{ContextualizedLogger, LoggingContext}
+import com.daml.platform.participant.util.HexOffset
+import com.daml.platform.testing.StreamConsumer
+import com.google.protobuf.ByteString
+import io.grpc.health.v1.health.{HealthCheckRequest, HealthCheckResponse}
+import io.grpc.stub.StreamObserver
+import scalaz.Tag
+import scalaz.syntax.tag._
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Failure
+import scala.util.control.NonFatal
+
+private[testtool] object SingleParticipantTestContext {
```

Comment:
It was the structure of the file beforehand. 

Wondering if I can somehow make git not believe I created this file, but only the trait one, will investigate so that we can keep the git history.

Reply:
Ok, had to keep a different name for the file with the trait, but managed to keep the history.